### DISCUSSION TITLE
Openapi3 importer: Generate endpoint docs.

### DISCUSSION THREAD

**Discussion by sebastianjonasson** [Kong/insomnia#3532] [Ideas]
Currently when importing an OpenAPI3 specification the Docs tab is almost empty. I would like the importer to generate docs for each endpoint.

Example doc generation:

### Endpoint summary here

GET /my/endpoi.nt

Endpoint description here.

### Path parameters
| name  | required  | in  |  description |
| ------------ | ------------ | ------------ | ------------ |
| userId  |  Yes | path  |  Id of the user |
| limit  |  No | query  |  Limit amount of items in response |

### Endpoint parameters
| name  | required  | in  |  description |
| ------------ | ------------ | ------------ | ------------ |
| fields  |  No | query  |  Fields to include in response |

### Responses
| Code  | Description  |
| ------------ | ------------ |
|  200 | OK  |
|  400  | Bad Request. Include userId in path  |
|  401  | Unauthorized  |

It could also include the following path item fields: requestBody, responses, and security.

I'm willing to implement this. But I want to feature to be approved first.

**Comment by sebastianjonasson** [comment]
@develohpanda

**Comment by nijikokun** [comment]
I'm not seeing any downsides to this at the moment, nor has anyone voiced any. Going to put the accepted label on it.

**Comment by sebastianjonasson** [comment]
Alright, sweet. Will start implementing this soon.
